apps/meteor/server/database/utils.ts

```diff
@@ -18,3 +18,21 @@ export const { db, client } = MongoInternals.defaultRemoteCollectionDriver().mon
 export const shouldRetryTransaction = (e: unknown): boolean =>
 	(e as MongoError)?.errorLabels?.includes('UnknownTransactionCommitResult') ||
 	(e as MongoError)?.errorLabels?.includes('TransientTransactionError');
+
+export const wrapInSessionTransaction =
+	<T extends Array<unknown>, U>(curriedCallback: (session: ClientSession) => (...args: T) => U) =>
+	async (...args: T): Promise<Awaited<U>> => {
+		const session = client.startSession();
+		try {
+			session.startTransaction();
+			const result = await curriedCallback(session).apply(this, args);
+			await session.commitTransaction();
+			return result;
+		} catch (error) {
```

Comment:
Mongo recommends transactions to be retried. Transient transaction errors could cause a rollback even when the tx is valid (a conflict with another op or something).

Also remember that, the longer the transaction spans, the easier it's for it to conflict with something.

Reply:
So I commented this with @sampaiodiego, and I actually had implemented retry calls for it at some point. But as you've seen there's so many side effects thrown around in inner functions, that I thought this would do more harm than good.

Reply:
So I fixed the issue with sideEffects, but we decided to not retry this transaction and show an error asking to try again. If another operation happens to the user while this method runs for example, it should probably have been aborted anyways